sql/rpl_slave.cc

```diff
@@ -5149,6 +5149,12 @@ static bool coord_handle_partial_binlogged_transaction(Relay_log_info *rli,
     ((Query_log_event*) begin_event)->db= "";
     begin_event->common_header->data_written= 0;
     begin_event->server_id= ev->server_id;
+    /*
+      This event is not generated on master and is only specific to replicas.
+      So, we don't want this BEGIN query to respect MASTER_DELAY.
+      Make the timestamp to be same as that of the event which triggered this.
+    */
```

Comment:
Optional: while it is clear that this event is not generated on master, can you please add that this is generated internally to handle FORMAT_DESCRIPTION_EVENT.. (or something on that lines)